toolkit/primitives/block-production-log/src/lib.rs

```diff
@@ -30,11 +30,29 @@ impl IsFatalError for InherentError {
 #[cfg(feature = "std")]
 #[derive(Debug)]
 pub struct BlockAuthorInherentProvider<Author> {
-	pub author: Author,
+	pub author: Option<Author>,
 }
 
 #[cfg(feature = "std")]
 impl<Author> BlockAuthorInherentProvider<Author> {
+	pub fn new_if_pallet_present<C, Member, Block>(
+		client: &C,
+		parent_hash: Block::Hash,
+	) -> Result<Self, Box<dyn Error + Send + Sync>>
+	where
+		Member: Decode,
+		Block: BlockT,
+		C: ProvideRuntimeApi<Block>,
+		C::Api: BlockProductionLogApi<Block, Member>,
+		Author: From<Member>,
+	{
+		let version = client.runtime_api().version(parent_hash)?;
+		if version.has_api_with(&<dyn BlockProductionLogApi<Block, Member>>::ID, |_| true) {
```

Comment:
there is shorter version with `has_api`, when we do not need to provide `with` predicate